src/engine/QueryPlanner.cpp

```diff
@@ -107,26 +184,73 @@ QueryExecutionTree QueryPlanner::createExecutionTree(ParsedQuery& pq) const {
           // the efficiency of the union operation is not dependent on the
           // sorting of the inputs and its position is fixed so it does not
           // need to be part of the optimization of the child.
-          SubtreePlan* left = &patternPlans[child->_childGraphPatterns[0]->_id];
-          SubtreePlan* right =
+          const SubtreePlan* left =
+              &patternPlans[child->_childGraphPatterns[0]->_id];
+          const SubtreePlan* right =
               &patternPlans[child->_childGraphPatterns[1]->_id];
 
           // create a new subtree plan
-          unionPlans.emplace_back(_qec);
+          childPlanStorage.emplace_back(_qec);
           std::shared_ptr<Operation> unionOp(
-              new Union(_qec, left->_qet, right->_qet));
-          QueryExecutionTree& tree = *unionPlans.back()._qet.get();
+              std::make_shared<Union>(_qec, left->_qet, right->_qet));
+          QueryExecutionTree& tree = *childPlanStorage.back()._qet.get();
```

Comment:
childPlanStorage stores `SubtreePlans`. The reference instead of a shared_ptr is a microoptimization, I guess, as we know that the shared_ptr will not be deleted and the storage is not freed.  `unionPlans` was renamed to reflect that it also stores `SubtreePlans` for `TransitivePath` operations (instead of just storing them for `Union` operations).